src/main/java/seedu/address/logic/commands/EditCommissionCommand.java

```diff
@@ -0,0 +1,275 @@
+        requireNonNull(model);
+
+        if (!model.hasSelectedCustomer()) {
+            throw new CommandException(Messages.MESSAGE_NO_ACTIVE_CUSTOMER);
+        }
+        Customer selectedCustomer = model.getSelectedCustomer().getValue();
+
+        List<Commission> lastShownList = model.getFilteredCommissionList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_COMMISSION_DISPLAYED_INDEX);
+        }
+
+        Commission commissionToEdit = lastShownList.get(index.getZeroBased());
+        Commission editedCommission = createEditedCommission(commissionToEdit, editCommissionDescriptor);
+
+        if (!commissionToEdit.isSameCommission(editedCommission) && selectedCustomer.hasCommission(editedCommission)) {
+            throw new CommandException(MESSAGE_DUPLICATE_COMMISSION);
+        }
+
+        selectedCustomer.setCommission(commissionToEdit, editedCommission);
+
+        model.selectCommission(editedCommission);
+
+        model.updateFilteredCommissionList(PREDICATE_SHOW_ALL_COMMISSIONS);
+        model.selectTab(GuiTab.COMMISSION);
+        return new CommandResult(String.format(MESSAGE_EDIT_COMMISSION_SUCCESS, editedCommission));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditCommissionCommand)) {
+            return false;
+        }
+
+        // state check
+        EditCommissionCommand e = (EditCommissionCommand) other;
+        return index.equals(e.index)
+                && editCommissionDescriptor.equals(e.editCommissionDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the customer with. Each non-empty field value will replace the
+     * corresponding field value of the customer.
+     */
+    public static class EditCommissionDescriptor {
+        private Title title;
+        private Fee fee;
+        private Deadline deadline;
+        private CompletionStatus completionStatus;
+        private Description description;
+        private Set<Tag> tags;
+
+        public EditCommissionDescriptor() {
+        }
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditCommissionDescriptor(EditCommissionDescriptor toCopy) {
+            setTitle(toCopy.title);
+            setFee(toCopy.fee);
+            setDeadline(toCopy.deadline);
+            setCompletionStatus(toCopy.completionStatus);
+            setTags(toCopy.tags);
+            setDescription(toCopy.description);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(title, fee, deadline, completionStatus, tags);
```

Comment:
Hm, just wondering if this would be an issue- by your current implementation, wanting to change only the description would not be permitted as AB would detect that as no fields being edited.

ie: trying to execute `editcom 1 p/Paint forest with blinding sunlight`, AB complains: `At least one field to edit must be provided.`

Reply:
oops missed this out - fixed in [fffd608](https://github.com/AY2223S1-CS2103T-W11-3/tp/pull/109/commits/fffd60857ef871c83169b6c03e5f873f2699eaca) - thanks!